ArduPlane/mode.h

```diff
@@ -296,17 +296,32 @@ class ModeGuided : public Mode
     // handle a guided target request from GCS
     bool handle_guided_request(Location target_loc) override;
 
+    // handle a guided path following request
+    bool handle_guided_path_request(Location position_on_path, Vector2f unit_path_tangent, const float path_curvature, const bool direction_is_ccw);
+
     void set_radius_and_direction(const float radius, const bool direction_is_ccw);
 
     void update_target_altitude() override;
 
 protected:
 
+    enum class SubMode: uint8_t {
+        WP,
+        Path
+    };
+
     bool _enter() override;
     bool _pre_arm_checks(size_t buflen, char *buffer) const override { return true; }
 
 private:
+
+    SubMode _guided_mode;
```

Comment:
can you check that the vehicle always ends up in the WP sub mode when it is first switched to Guided mode?  So it probably always does the first time the vehicle is switched to Guided but it may not if the vehicle is in Path sub mode (in Guided), then switched to another mode, then back to Guided.

Reply:
Do you think it's desireable for guided mode to preserve any state when you switch out of it and back into guided? 

For example, if you had sent a heading command, switch to RTL for 0.5 seconds, and back to guided, it seems the vehicle should chose the default WP sub mode and drop a point where it enters guided rather than go back to the old stale heading command.

Reply:
> Do you think it's desireable for guided mode to preserve any state when you switch out of it and back into guided?
> 

This is why I added a HEADING_TYPE_DEFAULT mavlink option in #27428  - so you can explicitly switch to GUIDED_HEADING_NONE.